app/controllers/search_controller.rb

```diff
@@ -2,10 +2,14 @@
 
 class SearchController < ApplicationController
   def index
-    mail = params[:search][:email]
-
-    @matching_users = User.where.not(id: current_user.id).all
-    @matching_users = @matching_users.where(email: mail.strip) if mail.present?
+    permitted = params.require(:search).permit(:email)
+    raise ActionController::ParameterMissing if permitted.blank?
```

Comment:
This raise was only valuable for testing if email is blank. If there are no permitted values, the previous statement should raise. If you are happy to do a search where email = '', then you can probably remove this line